index.js

```diff
@@ -10,86 +10,129 @@ if ("SERVER_LOGGER" in process.env) {
+        "version": process.env.npm_package_version || ""
+      },
+      "externalDocs": {
+        "url": "https://github.com/tobinbradley/dirt-simple-postgis-http-api",
+        "description": "Source code on Github"
+      },
+      "schemes": [
+        "http",
+        "https"
+      ],
+      "tags": [{
+        "name": "api",
+        "description": "code related end-points"
+      }, {
+        "name": "feature",
+        "description": "features in common formats for direct mapping."
+      }, {
+        "name": "meta",
+        "description": "meta information for tables and views."
+      }]
+    }
+  })
 
-// OPTIONAL RATE LIMITER
-if ("RATE_MAX" in process.env) {
-  fastify.register(import('@fastify/rate-limit'), {
-    max: process.env.RATE_MAX,
-    timeWindow: '1 minute'
+  // ADD ROUTES
+  fastify.register(require('@fastify/autoload'), {
+    dir: path.join(__dirname, 'routes')
   })
-}
 
-// INITIALIZE SWAGGER
-fastify.register(require('@fastify/swagger'), {
-  exposeRoute: true,
-  routePrefix: '/',
-  swagger: {
-    "info": {
-      "title": "Dirt-Simple PostGIS HTTP API",
-      "description": "The Dirt-Simple PostGIS HTTP API is an easy way to expose geospatial functionality to your applications. It takes simple requests over HTTP and returns JSON, JSONP, or protobuf (Mapbox Vector Tile) to the requester. Although the focus of the project has generally been on exposing PostGIS functionality to web apps, you can use the framework to make an API to any database.",
-      "version": process.env.npm_package_version || ""
-    },
-    "externalDocs": {
-      "url": "https://github.com/tobinbradley/dirt-simple-postgis-http-api",
-      "description": "Source code on Github"
-    },
-    "schemes": [
-      "http",
-      "https"
-    ],
-    "tags": [{
-      "name": "api",
-      "description": "code related end-points"
-    }, {
-      "name": "feature",
-      "description": "features in common formats for direct mapping."
-    }, {
-      "name": "meta",
-      "description": "meta information for tables and views."
-    }]
-  }
-})
+  fastify.get('/health-check', { logLevel: 'warn' }, (request, reply) => {
+    reply.send("healthy");
+  })
 
-// ADD ROUTES
-fastify.register(require('@fastify/autoload'), {
-  dir: path.join(__dirname, 'routes')
-})
 
-// LAUNCH SERVER
-fastify.listen({port: process.env.SERVER_PORT || 3000, host: process.env.SERVER_HOST || '0.0.0.0'}, (err, address) => {
-  if (err) {
-    console.log(err)
-    process.exit(1)
-  }
-  console.info(`Server listening on ${address}`)
-})
+  return fastify
+}
+
+build()
+  .then(fastify => // LAUNCH SERVER
+    fastify.listen({ port: process.env.SERVER_PORT || 3000, host: process.env.SERVER_HOST || '0.0.0.0' }, (err, address) => {
+      if (err) {
+        console.log(err)
+        process.exit(1)
+      }
+      console.info(`Server listening on ${address}`)
+    }))
+  .catch(console.log)

```

Comment:
Could we maybe add the error to logs

Reply:
If t this error occurs, most likely we didn't import fastify successfully and so the logger has not been initialized